fabric-networking-v0/src/main/java/net/fabricmc/fabric/api/network/handshake/HandshakeContext.java

```diff
@@ -13,13 +13,19 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package net.fabricmc.fabric.api.network.handshake;
 
-package net.fabricmc.fabric.api.network;
+import java.util.Collection;
+import java.util.List;
 
-import net.fabricmc.fabric.impl.network.ServerRequiresModRegistryImpl;
+import net.fabricmc.loader.api.FabricLoader;
 
-public interface ServerRequiresModRegistry {
-    public static final ServerRequiresModRegistry INSTANCE = ServerRequiresModRegistryImpl.INSTANCE;
-
-    public void requireModOfVersion(String modid, String version);
+public interface HandshakeContext {
+    public FabricLoader getLoader();
+    
+    public Collection<String> getClientModIDs();
+    
+    public void fail(FailureReason reason);
+    
+    public Collection<FailureReason> getFailureReasons();
```

Comment:
Don't need public modifiers;
Also consider return a list in which the reasons are listed in encounter order